packages/commerce-server/src/record-new-purchase.ts

```diff
@@ -99,52 +88,52 @@ export async function recordNewPurchase(
     getMerchantProduct,
   } = getSdk()
 
-  const {stripeCtx} = options
-
-  const purchaseInfo = await stripeData({checkoutSessionId, stripeCtx})
+  const purchaseInfo = await options.paymentProvider.getPurchaseInfo(
+    checkoutSessionId,
+  )
 
   const {
-    stripeCustomerId,
+    customerIdentifier,
     email,
     name,
-    stripeProductId,
-    stripeChargeId,
-    stripeCouponId,
+    productIdentifier,
+    chargeIdentifier,
+    couponIdentifier,
     quantity,
-    stripeChargeAmount,
+    chargeAmount,
     metadata,
```

Comment:
account for some variable renaming from `purchaseInfo` because we are trying this as provider agnostic info